diff --git a/components/Page.astro b/components/Page.astro
index 8e85641d2e17d25b5e3f0793c32f369fe1b7d407..27cda4c846912f830c0aef33121b0e201ee73beb 100644
--- a/components/Page.astro
+++ b/components/Page.astro
@@ -6,12 +6,14 @@ import StarlightPage, { type StarlightPageProps as Props } from '@astrojs/starli
   <slot />
 </StarlightPage>

+<script>document.documentElement.setAttribute('data-blog-page', '')</script>
+
 <style>
-  :global([data-has-toc] :is(.right-sidebar-panel, mobile-starlight-toc)) {
+  :global([data-blog-page][data-has-toc] :is(.right-sidebar-panel, mobile-starlight-toc)) {
     display: none;
   }

-  :global([data-has-toc] .main-frame) {
+  :global([data-blog-page][data-has-toc] .main-frame) {
     padding-top: var(--sl-nav-height);
   }
 </style>
diff --git a/routes/Blog.astro b/routes/Blog.astro
index e81954e0970e6a5c0fcd1d8123b96cf33caa0055..3141e234084014b4b10d1f8ecdc6f62164b21960 100644
--- a/routes/Blog.astro
+++ b/routes/Blog.astro
@@ -29,11 +29,11 @@ const pageProps = getPageProps(getBlogTitle(locale))
 </Page>

 <style>
-  :global(.content-panel:first-of-type) {
+  :global([data-blog-page] .content-panel:first-of-type) {
     display: none;
   }

-  :global(.content-panel:nth-of-type(2)) {
+  :global([data-blog-page] .content-panel:nth-of-type(2)) {
     border-top: none;
   }
 </style>
